enhancements/monitoring/client-cert-scraping.md

```diff
@@ -0,0 +1,132 @@
+---
+title: client-cert-base-metric-scraping
+authors:
+  - "@deads2k"
+reviewers:
+  - "@sur"
+approvers:
+  - "@sur"
+creation-date: 2021-03-18
+last-updated: 2021-03-18
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Client Cert Base Metrics Scraping
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Monitoring needs to be reliable and is the very useful when trying to debug clusters in an already degraded state.
+We want to ensure that metrics scraping can always work if the scraper can reach the target, even if the kube-apiserver
+is unavailable or unreachable.
+To do this, we will combine a local authorizer (already merged in many binaries and the rbac-proxy) and client-cert based
+authentication to have a fully local authentication and authorization path for scraper targets.
+
+## Motivation
+
+If networking (or part of networking) is down and a scraper target cannot reach the kube-apiserver to verify a token
+and a subjectaccessreview, then the metrics scraper can be rejected.
+The subjectaccessreview (authorization) is already largely addressed, but service account tokens are still used
+for scraping targets.
+Tokens require an external network call that we can avoid by using client certificates.
```

Comment:
question: as tokens are signed, wouldn't a local signature check suffer?

Reply:
as an after thought, potentially a signature-only check in addition to scoped tokens would potentially suffice?
At least the advantage would be that client-side no additional cert management would be necessary.

Reply:
> question: as tokens are signed, wouldn't a local signature check suffice?

No. In general tokens are not signed.  ServiceAccount tokens in particular are, *but* if you only check locally, then the serviceaccount tokens become irrevocable.  It is important to be able to revoke tokens.